src/main/java/org/testng/xml/XmlTest.java

```diff
@@ -582,7 +582,7 @@ public String toXml(String indent) {
   @Override
   public String toString() {
 //    return toXml("");
-    StringBuffer result = new StringBuffer("[Test: \"" + m_name + "\"")
+    StringBuilder result = new StringBuilder("[Test: \"" + m_name + "\"")
```

Comment:
Avoid String#format, which is super slow (as in, really, really slow). If we're just adding straight `CharSequences`, there's really no good reason to use it since you're really not formatting.

Other than that, I don't really care between `StringBuilder` or `StringBuffer`, either is fine.